### PR TITLE
Implementing pangram.

### PR DESCRIPTION
**Pull request by Is0metry**
description to come

### CODE REVIEW

**Comment by patricksjackson** [comment]
This is pretty good :)

The test suite is complete, and the implementation is good enough I won't bother nitpicking it.

What you'll need to do before I can merge it is finish matching the style in this repo and adding it to our build scripts. So a short list of things you'll need to fix:

- Please don't re-format the config.json. Just add the lines necessary for this new problem.
- You'll need to add the name to the list in the root [CMakeLists.txt](https://github.com/exercism/cpp/blob/master/CMakeLists.txt), otherwise it won't build during our automated testing.
- Try to match the formatting of existing tests. Take a look at [Word Count](https://github.com/exercism/cpp/tree/master/exercises/word-count) in detail, it has similar requirements to this one. Specifically:
1. Formatting, specifically vertical spacing (blank lines).
2. Name of the `#define` in the header.
3. Placing the helper function in an anonymous namespace and not exposing it in the header.
4. Including only what you need in the header file, and moving the others to the cpp (`cctype`).

**Comment by Is0metry** [comment]
changes implemented :)

I didn't consciously reformat the config (I ran everything through the configlet which might've done it), so I rolled back to a previous version and just added the necessary lines. It **should** be back to normal again.

File: config.json [hunk 1]
```diff
@@ -423,6 +423,14 @@
         "pattern matching",
         "stacks"
       ]
+    },
+    {
+      "slug":"pangram",
```
Comment (patricksjackson) [inline config.json]: This needs a couple more fields (the `uuid`, which configlet can generate for you, `core` and `unlocked_by` (just leave the values the same default as the others.

File: exercises/pangram/example.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#ifndef PANGRAM_H
```
Comment (patricksjackson) [inline exercises/pangram/example.h]: `#if !defined(PANGRAM_H)` to match all the other problems.

File: exercises/pangram/example.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+#define PANGRAM_H
+#include <string>
+
+namespace pangram
+{
+    
+   bool is_pangram(std::string s);
```
Comment (patricksjackson) [inline exercises/pangram/example.h]: The repo convention is to not indent namespace contents. So this whole namespace should be:

```
namespace pangram
{

bool is_pangram(std::string s);

}
```

File: exercises/pangram/example.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#include "example.h"
+#include <cctype>
+namespace
+{
+    bool found_letter(char a, std::string s)
```
Comment (patricksjackson) [inline exercises/pangram/example.cpp]: Please add some extra vertical spacing (blank lines between the namespaces/functions/etc (see [word-count](https://github.com/exercism/cpp/blob/master/exercises/word-count/example.cpp))

And don't indent inside of the namespace block.

File: config.json (hunk 1 above)
Comment (Is0metry) [reply config.json]: What command do I run on the configlet to get it to add the `uuid`? I've tried `fmt` and `lint` and neither seems to be doing it.

**Comment by patricksjackson** [comment]
Thanks :) I pointed out a few more details.

Thanks for letting me know the configlet tool was the one that reformatted the json. I'd like to update the formatting of that with whatever configlet does, but that should be done in a separate commit.

(Also, still need the CMakeLists.txt update I mentioned, though I know you may be working on it :) )

**Comment by Is0metry** [comment]
That should be just about everything. Thanks for your patience. I'm still kinda new to this :).

File: config.json (hunk 1 above)
Comment (patricksjackson) [reply config.json]: I should clarify this - `configlet uuid` will generate the uuid string for you, but you'll need to edit the json manually

File: exercises/pangram/example.h [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#if !defined(PANGRAM_H)
+#define PANGRAM_H
+#include <string>
+
+namespace pangram
+{
+    
+bool is_pangram(std::string s);
```
Comment (patricksjackson) [inline exercises/pangram/example.h]: Actually please make the string arguments for both of these functions take `std::string const& s`.

I missed this on my first pass...however this and the config.json update are the only things left :)

Comment (Is0metry) [reply exercises/pangram/example.h]: latest push should have everything.

**Comment by patricksjackson** [comment]
Awesome! Thanks.

**Pull request merged**